chain/src/join_split.rs

```diff
@@ -1,47 +1,166 @@
-use std::io;
+use std::{fmt, io};
 use hash::{H256, H512};
-use ser::{Error, Stream, Reader, FixedArray_H256_2,
-	FixedArray_u8_296, FixedArray_u8_601_2};
+use hex::ToHex;
+use ser::{Error, Stream, Reader, CompactInteger};
+
+#[derive(Clone)]
+pub enum JoinSplitProof {
+	PHGR([u8; 296]),
+	Groth([u8; 192]),
+}
 
 #[derive(Debug, PartialEq, Default, Clone)]
-pub struct JointSplit {
-	pub descriptions: Vec<JointSplitDescription>,
+pub struct JoinSplit {
+	pub descriptions: Vec<JoinSplitDescription>,
 	pub pubkey: H256,
 	pub sig: H512,
 }
 
-#[derive(Debug, PartialEq, Default, Clone, Serializable, Deserializable)]
-pub struct JointSplitDescription {
+#[derive(Clone)]
+pub struct JoinSplitDescription {
 	pub value_pub_old: u64,
 	pub value_pub_new: u64,
-	pub anchor: H256,
-	pub nullifiers: FixedArray_H256_2,
-	pub commitments: FixedArray_H256_2,
-	pub ephemeral_key: H256,
-	pub random_seed: H256,
-	pub macs: FixedArray_H256_2,
-	pub zkproof: FixedArray_u8_296,
-	pub ciphertexts: FixedArray_u8_601_2,
+	pub anchor: [u8; 32],
+	pub nullifiers: [[u8; 32]; 2],
```

Comment:
why not using H256 instead of all [u8;32] ?

Reply:
`H*` have associations with some hashing for me. I.e. `fn sha256() -> H256`. I'm still not sure how we're going to use all these fields. Until then I'd prefer to leave them as blobs, not hashes